src/components/FlySelect/FlySelect.tsx

```diff
@@ -1,53 +1,71 @@
-import React, { Component } from 'react';
+import * as React from 'react';
 import classnames from 'classnames';
-import CheckSVG from '../../svg/checkmark--big.svg';
-import DownloadSmallSVG from '../../svg/download--small.svg';
-import ArrowRightSVG from '../../svg/arrow--right.svg';
-import PropTypes from 'prop-types';
-
-export default class FlySelect extends Component {
-	PropTypes = {
-		options: PropTypes.object,
-		optionGroups: PropTypes.object,
-		value: PropTypes.any,
-		onChange: PropTypes.func.isRequired,
-		placeholder: PropTypes.string,
-		emptyPlaceholder: PropTypes.string,
-		disabled: PropTypes.bool,
-		footerText: PropTypes.string,
-		footerOnClick: PropTypes.func,
-	};
-
-	constructor (props) {
+import CheckSVG from '../../svg/checkmark--big';
+import DownloadSmallSVG from '../../svg/download--small';
+import ArrowRightSVG from '../../svg/arrow--right';
+import IReactComponentProps from '../../common/structures/IReactComponentProps';
+
+interface IProps extends IReactComponentProps {
+
+	disabled?: boolean;
+	emptyPlaceholder?: string;
+	footerText?: string;
+	footerOnClick?: (...params: any[]) => any;
+	loadingOptionsPlaceholder?: any;
+	onChange: (...params: any[]) => any;
+	options?: any;
+	optionsLoader?: any;
+	optionGroups?: any;
+	placeholder?: string;
+	value?: any;
+
+}
+
+interface IState {
+
+	focus: boolean;
+	options: any;
+	open: boolean;
+	optionsLoaded: boolean | null;
+	value: any;
+	willClose: boolean;
+
+}
+
+export default class FlySelect extends React.Component<IProps, IState> {
+
+	private readonly __containerRef: React.RefObject<any>;
```

Comment:
I love seeing things like this in TypeScript 😃

Reply:
Agreed. I think it certainly does help both in terms of readability and avoiding coding mistakes.